public final void wait() throws InterruptedException
public final native void wait(long ms) throws InterruptedException
public final void wait(long ms, int nanos) throws InterruptedException


public final native void notify()
public final native void notifyAll()

If there is no waiting state, then throws InterruptedException is not there.
So, wait() method throws InterruptedException which is checked exception but notify() and notifyAll() methods do not throw InterruptedException.